components/mediators/oauth/org.wso2.carbon.identity.oauth.mediator/src/main/java/org/wso2/carbon/identity/oauth/mediator/config/xml/OAuthMediatorFactory.java

```diff
@@ -59,14 +60,21 @@ public Mediator createSpecificMediator(OMElement element, Properties properties)
         remoteServiceUrl = element.getAttribute(ATTR_NAME_SERVICE_EPR);
         if (remoteServiceUrl != null && remoteServiceUrl.getAttributeValue() != null) {
             mediator.setRemoteServiceUrl(remoteServiceUrl.getAttributeValue());
+        }else {
+            throw new MediatorException(
+                    "The 'remoteServiceUrl' attribute is required for the OAuth mediator");
         }
         username = element.getAttribute(ATTR_NAME_USERNAME);
         if(username != null && username.getAttributeValue() != null){

```

Comment:
Let us use proper parentheses to separate the logic.
what abt 
 (null != username) && (null != username.getAttributeValue())